src/base.jl

```diff
@@ -402,6 +402,27 @@ function add_component!(
     return
 end
 
+"""
+Add many components to the system at once.
+
+Throws ArgumentError if the component's name is already stored for its concrete type.
+Throws ArgumentError if any Component-specific rule is violated.
+Throws InvalidRange if any of the component's field values are outside of defined valid
+range.
+
+# Examples
+```julia
+sys = System(100.0)
+
+buses = [bus1, bus2, bus3]
+generators = [gen1, gen2, gen3]
+foreach(x -> add_component!(sys, x), Iterators.flatten((buses, generators)))
+```
+"""
+function add_components!(sys::System, components)
```

Comment:
@daniel-thom should I specify the type of `components`? If so, what do you think I should use?

Reply:
No. This is best. There is no way to specify any type of iterable.